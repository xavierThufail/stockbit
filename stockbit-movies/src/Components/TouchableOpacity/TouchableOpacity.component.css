.touchable {
  cursor: pointer;
}

.touchable:active {
  opacity: 0.6;
}