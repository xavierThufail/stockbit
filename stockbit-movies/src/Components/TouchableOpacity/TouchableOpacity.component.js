import React from 'react';

import './TouchableOpacity.component.css';
import config from './TouchableOpacity.config';

const TouchableOpacity = ({ children, onClick, style }) => (
  <div
    onClick={onClick}
    style={style}
    className='touchable'
  >
    {children}
  </div>
);

TouchableOpacity.propTypes = config.propTypes;
TouchableOpacity.displayName = config.displayName;

export default TouchableOpacity;
