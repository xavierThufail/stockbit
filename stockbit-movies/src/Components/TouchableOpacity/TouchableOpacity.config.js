import PropTypes from 'prop-types';

const propTypes = {
  style: PropTypes.object,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func
};

const displayName = 'TouchableOpacity';

export default {
  propTypes,
  displayName
};
