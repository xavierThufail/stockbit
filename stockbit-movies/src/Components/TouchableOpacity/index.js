export { default as TouchableOpacity } from './TouchableOpacity.component';
