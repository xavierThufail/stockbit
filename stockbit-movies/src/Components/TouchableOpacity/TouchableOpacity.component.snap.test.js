import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '../../matchMedia.mock';

import TouchableOpacity from './TouchableOpacity.component';

describe('TouchableOpacity component snap test', () => {
  it('Render TouchableOpacity, should match snapshot', () => {
    const onClick = jest.fn();
    const title = 'Click Here';

    const { container, getByText } = render(<TouchableOpacity onClick={onClick}>{title}</TouchableOpacity>);

    const btn = getByText(title);

    expect(btn).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
