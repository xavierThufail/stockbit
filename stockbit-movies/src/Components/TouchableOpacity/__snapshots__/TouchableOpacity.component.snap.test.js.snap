// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableOpacity component snap test Render TouchableOpacity, should match snapshot 1`] = `
<div>
  <div
    class="touchable"
  >
    Click Here
  </div>
</div>
`;
